example/glue/ScavengerDelegate.cpp

```diff
@@ -0,0 +1,222 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "omrcfg.h"
+
+#if defined(OMR_GC_SCAVENGER_DELEGATE)
+
+#include "ScavengerDelegate.hpp"
+
+#include "j9nongenerated.h"
+#include "modronbase.h"
+#if defined(OMR_GC_MODRON_CONCURRENT_MARK)
+#include "CardTable.hpp"
+#endif /* OMR_GC_MODRON_CONCURRENT_MARK */
+#include "CollectorLanguageInterfaceImpl.hpp"
+#if defined(OMR_GC_MODRON_CONCURRENT_MARK)
+#include "ConcurrentSafepointCallback.hpp"
+#endif /* OMR_GC_MODRON_CONCURRENT_MARK */
+#if defined(OMR_GC_MODRON_COMPACTION)
+#include "CompactScheme.hpp"
+#endif /* OMR_GC_MODRON_COMPACTION */
+#include "EnvironmentStandard.hpp"
+#include "ForwardedHeader.hpp"
+#include "GCExtensionsBase.hpp"
+#include "HeapLinkedFreeHeader.hpp"
+#include "MarkingScheme.hpp"
+#include "MemorySubSpaceSemiSpace.hpp"
+#include "MixedObjectScanner.hpp"
+#include "mminitcore.h"
+#include "objectdescription.h"
+#include "ObjectIterator.hpp"
+#include "ObjectModel.hpp"
+#include "omr.h"
+#include "omrExampleVM.hpp"
+#include "omrvm.h"
+#include "OMRVMInterface.hpp"
+#include "ParallelGlobalGC.hpp"
+#include "Scavenger.hpp"
+#include "SlotObject.hpp"
+
+void
+MM_ScavengerDelegate::masterSetupForGC(MM_EnvironmentBase *env)
```

Comment:
The implementation of this file was taken from the example's CollectorLanguageInterfaceImpl, with the following changes:

* wrapped body in ifdef(OMR_GC_SCAVENGER_DELEGATE)
* renamed CLI to MM_ScavengerDelegate
* deleted virtual keyword

Reply:
Can we also remove the remaining `*_` prefixes from the function names? They too seem superfluous at this point...